blockchain/chain.go

```diff
@@ -1665,6 +1665,35 @@ func (b *BlockChain) locateHeaders(locator BlockLocator, hashStop *chainhash.Has
 	return headers
 }
 
+// ReconsiderBlock takes a block hash and allows it to be revalidated.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) ReconsiderBlock(hash string) error {
```

Comment:
I would make the parameter here a `chainhash.Hash` and let the caller turn the string into the chainhash.